pkg/awsutils/awsutils.go

```diff
@@ -740,8 +741,40 @@ func (cache *EC2InstanceMetadataCache) awsGetFreeDeviceNumber() (int, error) {
 
 // AllocENI creates an ENI and attaches it to the instance
 // returns: newly created ENI ID
-func (cache *EC2InstanceMetadataCache) AllocENI(useCustomCfg bool, sg []*string, subnet string, numIPs int) (string, error) {
-	eniID, err := cache.createENI(useCustomCfg, sg, subnet, numIPs)
+func (cache *EC2InstanceMetadataCache) AllocENI(useCustomCfg bool, sg []*string, subnets []*string, numIPs int) (string, error) {
+	describeSubnetInput := &ec2.DescribeSubnetsInput{
```

Comment:
This should only be done for custom networking, as in the normal case, we do not want to make a Describe call